src/packages/documents/documents/entity-actions/culture-and-hostnames/repository/culture-and-hostnames.repository.ts

```diff
@@ -0,0 +1,51 @@
+import { UmbDocumentCultureAndHostnamesServerDataSource } from './culture-and-hostnames.server.data.js';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import { UmbBaseController } from '@umbraco-cms/backoffice/class-api';
+import type { UmbNotificationContext } from '@umbraco-cms/backoffice/notification';
+import { UMB_NOTIFICATION_CONTEXT } from '@umbraco-cms/backoffice/notification';
+import type { UmbApi } from '@umbraco-cms/backoffice/extension-api';
+import type { DomainsPresentationModelBaseModel } from '@umbraco-cms/backoffice/backend-api';
+
+export class UmbDocumentCultureAndHostnamesRepository extends UmbBaseController implements UmbApi {
+	#init!: Promise<unknown>;
+
+	#dataSource: UmbDocumentCultureAndHostnamesServerDataSource;
+
+	#notificationContext?: UmbNotificationContext;
```

Comment:
Consider inferring the type directly from the token:

```suggestion
	#notificationContext?: typeof UMB_NOTIFICATION_CONTEXT.TYPE;
```